### PR TITLE
Carousel: Use existing gallery image to get background colour

### PR DESCRIPTION
**Pull request by glendaviesnz**
#### Changes proposed in this Pull Request:
* Adds a cross origin setting to fix security exception on background colour extraction from canvas
* Uses the existing image in the gallery for the generation of the background colour
* Sets background image on the swiper virtual start and finish slides

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Checkout PR to local test env and enable the carousel slide background colour option under jetpack 'Writing' settings
- Add a gallery and open in frontend and make sure the background colour shows correctly on all slides looping forwards and backwards
- Also test on WPcom and Atomic

### CODE REVIEW

**Comment by stacimc** [comment]
I pushed a commit that has the background colors loading in on wpcom now, although I'm afraid I'm not entirely sure why it's happening. We now call `calculateSlideBackgroundCss` on `image.onload` regardless of whether `isLoaded` already returned true. It does not work if we remove this entirely, however:

```
                        if ( isLoaded ) {
				calculateSlideBackgroundCss( slideEl, image );
			}
```

It seems it's calculating a (0, 0, 0) rgb in the first pass and correctly calculating it on the second one; I wonder if some of our assumptions about `isLoaded` may be wrong.

I have confirmed via logging that this is calculating rgb twice per slide when the carousel loads, but not recalculating thereafter.

Now the problem is that we're back to the original issue where the first & last images sometimes lose their background color when swiping. I think this is due to swiper creating duplicated slides for loop mode.

File: projects/plugins/jetpack/modules/carousel/jetpack-carousel.js
Comment (stacimc) [inline projects/plugins/jetpack/modules/carousel/jetpack-carousel.js:46]: I think we should pull #20485 into this PR since we're already setting the `crossOrigin` here so we can test. This should be updated with a try/catch per the comment on that PR https://github.com/Automattic/jetpack/pull/20485#discussion_r679593376

File: projects/plugins/jetpack/modules/carousel/jetpack-carousel.js
Comment (stacimc) [inline projects/plugins/jetpack/modules/carousel/jetpack-carousel.js:41]: It's just occurred to me that this is not a good default rgb if the carousel is in Light Mode; the images will then get a gray background because of the alpha gradient. We could add a check for the `jp-carousel-light` class and return white 🤔

**Comment by stacimc** [comment]
This is now working well for me on wpcom and jetpack sites. In order to get the background colors working correctly I did have to go back to the original less optimized implementation, so I think it's worth another round of performance testing. So far this appears to be running pretty smoothly for me.

**Review by ramonjd — DISMISSED** [review]

**Comment by ramonjd** [comment]
I tested in Firefox, Chrome, IE, and Safari on Jurassic.tube and WPCOM.

For Atomic I created an ephemeral site with Jetpack beta and ran this feature branch.

On desktop browsers the background appears as expected. I can't detect any performance issues when throttling CPU/network. ✅ 

On mobile device (testing on Browserstack), the background appeared as expected on Safari, Chrome and Firefox on iOS and Android.✅ 
 
I don't have any comment as to use of `imgEl.crossOrigin = 'Anonymous';`. I'm not sure we could use `use-credentials` anyway if images are coming from the CDN (?)

We do use `crossOrigin = 'Anonymous'` elsewhere on WPCOM I see.

I think I'll defer to the Jetpack crew for their opinion on this.

This PR needs a rebase I think as well.

**Comment by samiff** [comment]
> I think I'll defer to the Jetpack crew for their opinion on this.

Given the timing here, I think it'd be best to revert the colorized background feature and shoot again for the next release cycle (JP 10.1) where we have more time for review and testing.

@glendaviesnz @stacimc @ramonjd @andrewserong 

Could one of you work on getting a revert PR ready, and once that is merged to master I'll take care of reverting on the 10.0 release branch, thanks! I believe that https://github.com/Automattic/jetpack/pull/20377 was the main PR for the feature.

**Comment by apeatling** [comment]
Closing this one out because it's causing all sorts of new issues that we don't have the bandwidth to address right now. We will revisit, likely in core.

**Pull request closed without merging**
